### PR TITLE
fix service workers

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #3313

### CODE REVIEW

File: assets/panel/panel.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ DebuggerPanel.prototype = {
     } = await this.panelWin.Debugger.bootstrap({
       threadClient: this.toolbox.threadClient,
       tabTarget: this.toolbox.target,
-      debuggerClient: this.toolbox.target._client,
+      debuggerClient: this.toolbox.target.client,
```
Comment (jasonLaster) [inline assets/panel/panel.js]: this is already in m-c

File: src/client/firefox.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export async function onConnect(connection: any, actions: Object) {
   // bfcache) so explicity fire `newSource` events for all returned
   // sources.
   const sources = await clientCommands.fetchSources();
-  actions.connect(tabTarget._form.url);
+  actions.connect(tabTarget.url);
```
Comment (jasonLaster) [inline src/client/firefox.js]: there's a getter on the targets that we can use

**Comment by clarkbw** [comment]
This fixes #3368

File: src/client/firefox/commands.js [hunk 3]
```diff
@@ -160,6 +160,10 @@ type EvaluateParam = {
 
 function evaluate(script: Script, { frameId }: EvaluateParam) {
   const params = frameId ? { frameActor: frameId } : {};
+  if (!tabTarget || !tabTarget.activeConsole) {
+    return Promise.resolve();
```
Comment (bgrins) [inline src/client/firefox/commands.js]: What's the case when this happens? Is this expected until we fix some issue on the target? Perhaps we should console.warn in this case

Comment (bgrins) [inline src/client/firefox/commands.js]: A comment above this condition explaining would be helpful as well

Comment (jasonLaster) [inline src/client/firefox/commands.js]: Just added an issue for attaching to the console synchronously.

https://bugzilla.mozilla.org/show_bug.cgi?id=1382302

**Pull request merged**